core/include/rmutex.h

```diff
@@ -54,18 +53,17 @@ typedef struct rmutex_t {
     /**
      * @brief   Owner thread of the mutex.
      * @details Owner is written by the mutex holder, and read
-     *          concurrently to ensure consistency,
-     *          atomic_int_least16_t is used. Note @ref kernel_pid_t is an int16
+     *          in critical section to ensure consistency.
      * @internal
      */
-    atomic_int_least16_t owner;
```

Comment:
If `atomic_uint_fast16_t` instead of `atomic_int_least16_t` would be used, this should work without the compiler having to insert a library call (which simple calls `irq_disable()` before the access and `irq_restore()` afterwards). I think that ARM can only provide atomic access to memory if it is word-sized and aligned on word-boundaries.

Reply:
unfortunately it doesn't, still triggering the compiler warning.